# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/CanvasRenderingContext2D.webidl
# Commit: 1c98db62bb1a
# Full Hash: 1c98db62bb1a20d26f812d28d86672158450e71c
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-03-28 21:36:34
# Regressor Bug: 1887729
# File Overlap Count: 2
# Description:
#   Bug 1887729 - Implement context lost/restored support for CanvasRenderingContext2D. r=webidl,gfx-reviewers,smaug,lsalzman
#   
#   Remote canvas can run in the GPU process, and if the GPU process
#   crashes, we need to notify the application using canvas. Historically we
#   just failed, and the application may have been able to continue drawing
# ==============================================================================

diff -r 315bfe14fe3f -r 1c98db62bb1a dom/webidl/CanvasRenderingContext2D.webidl
--- a/dom/webidl/CanvasRenderingContext2D.webidl	Thu Mar 28 14:21:58 2024 +0000
+++ b/dom/webidl/CanvasRenderingContext2D.webidl	Thu Mar 28 14:50:20 2024 +0000
@@ -151,6 +151,7 @@
   undefined save(); // push state on state stack
   undefined restore(); // pop state stack and restore state
   undefined reset(); // reset the rendering context to its default state
+  boolean isContextLost(); // return whether context is lost
 };
 
 interface mixin CanvasTransform {